core/src/test/java/com/klaytn/caver/ipfs/IPFSTest.java

```diff
@@ -81,6 +81,9 @@ public void toHex() {
 
     @Test
     public void addFile() throws IOException {
+        if (System.getenv("LOCAL_JVM") == null) {
```

Comment:
what does this code mean? 
When I wants to test, do I need to set the LOCAL_JVM environment variable? 
How was it intended to work in the Circle CI environment?

Reply:
@sirano11 
Hi sirano :)
I added new Environment Variable "LOCAL_JVM" to circle ci config.
This env-variable is used to specify the environment test is being executed.

If `LOCAL_JVM` is set, it means "JVM" environment.
Else it means "Android" environment.

```
jobs:
  build:
    <<: *machine_ubuntu
    steps:
      - checkout
      - *pull_klaytn_image
      - *start_test_network
      - *check_test_network
      - run:
          name: "Check rc version"
          command: |
            if [ -z "$CIRCLE_TAG" ]; then
              echo "this is not RC version"
            else
              sed -i "s/version '.*'/version '${CIRCLE_TAG:1}'/" build.gradle
              awk '/version '\''/' build.gradle
            fi
            export LOCAL_JVM=true // HERE, but this may not an appropriate position. I think I should move this statement outside of `Check rc version` block.
```

1. If a test is running on a machine which has "LOCAL_JVM" environment variable and it contains `if` statement about `LOCAL_JVM` and it's condition is true, it means "This test case can only be running on JVM, not an Android Emulator."
2. If a test is running on a machine which has no `if` statement about "LOCAL_JVM" environment variable, it means "This test case can be running on both JVM and Android Emulator"

So with this `LOCAL_JVM`, we can make an android emulator skip test cases which can be only ran in JVM.

Reply:
So, How can i test a unit test in local? 
I think this code is focused only circle ci system.